src/h5web/visualizations/heatmap/HeatmapVis.tsx

```diff
@@ -11,12 +14,32 @@ interface Props {
 function HeatmapVis(props: Props): JSX.Element {
   const { dims, data } = props;
   const textureData = useMemo(() => computeTextureData(data), [data]);
+  const [leftAreaRef, leftAreaBounds] = useMeasure();
+  const [bottomAreaRef, bottomAreaBounds] = useMeasure();
```

Comment:
Okay so I think I figured out the perf issue on resize: these two `useMeasure` hooks mean that the canvas gets re-rendered on every resize twice more. You could try to move these inside the `IndexAxis` component, along with the wrapper `div`'s. This also avoids duplicating the logic.

Reply:
:fishing_pole_and_fish: